homeassistant/components/tuya/vacuum.py

```diff
@@ -95,14 +95,11 @@ def __init__(self, device: TuyaDevice, device_manager: TuyaDeviceManager) -> Non
         if self.find_dpcode(DPCode.PAUSE, prefer_function=True):
             self._supported_features |= SUPPORT_PAUSE
 
-        if (
-            self.find_dpcode(DPCode.SWITCH_CHARGE, prefer_function=True)
-            or TUYA_MODE_RETURN_HOME
-            in EnumTypeData.from_json(
-                DPCode.MODE, device.status_range.get(DPCode.MODE).values
-            ).range
-        ):
+        if self.find_dpcode(DPCode.SWITCH_CHARGE, prefer_function=True):
             self._supported_features |= SUPPORT_RETURN_HOME
+        elif mode := self.find_dpcode(DPCode.MODE, dptype=DPType.ENUM):
+            if TUYA_MODE_RETURN_HOME in mode.range:
```

Comment:
```suggestion
        elif (enum_type := self.find_dpcode(DPCode.MODE, dptype=DPType.ENUM, prefer_function=True) and TUYA_MODE_RETURN_HOME in enum_type.range:
```

(this needs to  be formatted with black)

Reply:
Thanks!